src/ns-support.c

```diff
@@ -345,6 +349,20 @@ void sc_create_or_join_ns_group(struct sc_ns_group *group)
 		if (prctl(PR_SET_PDEATHSIG, SIGINT, 0, 0, 0) < 0) {
 			die("cannot set parent process death notification signal to SIGINT");
 		}
+		// Check that parent process is still alive. If this is the case then
+		// we can reliably rely on the PR_SET_PDEATHSIG signal to wake us up
+		// from eventfd_read() below.
+		debug("ensuring that parent process is still alive");
+		if (kill(parent, 0) < 0) {
+			switch (errno) {
+			case ESRCH:
+				debug("parent process has already terminated");
+				abort();
+			default:
+				die("cannot ensure that parent process is still alive");
+				break;
+			}
+		}
```

Comment:
After talking on IRC, the intent of this is simply an early warning system if the parent dies right away and that an in progress PR will implement a timeout. In this light, this PR is fine to land as is-- in the normal case the pid won't be reused and we can bail fast but if the pid is reused, that process won't send the signal anyway and the timeout expire and we can cleanup then.

+1 provided that a comment is added regarding pid reuse and that we'll use a timeout to handle that.